packages/client/src/routes/Login/Login.tsx

```diff
@@ -0,0 +1,52 @@
+import { FunctionalComponent, h } from 'preact';
+import { useCallback, useContext, useState } from 'preact/hooks';
+import { getConfig } from '@api/config';
+import style from './style.css';
+import authContext from 'src/components/authContext';
+
+interface ILogin {
+	onSuccess?: () => void;
+	onFail?: () => void;
+}
+
+const Login: FunctionalComponent<ILogin> = ({ onFail }) => {
+	const { setAuthHeader } = useContext(authContext);
+
+	const [username, setUsername] = useState<string>('');
+	const [password, setPassword] = useState<string>('');
+
+	const handleUsernameChange: h.JSX.GenericEventHandler<HTMLInputElement> = useCallback(
+		(event) => setUsername((event.target as HTMLInputElement).value),
+		[]
+	);
+	const handlePasswordChange: h.JSX.GenericEventHandler<HTMLInputElement> = useCallback(
+		(event) => setPassword((event.target as HTMLInputElement).value),
+		[]
+	);
+
+	const handleSubmit: h.JSX.MouseEventHandler<HTMLButtonElement> = useCallback(async () => {
+		// const encoded = Buffer.from(`${username}:${password}`).toString('base64');
+		const encoded = btoa(`${username}:${password}`);
+		const result = await getConfig(encoded);
+		if (!result) {
+			return onFail?.();
+		}
+		setAuthHeader(encoded);
+	}, [username, password]);
```

Comment:
Ok cool. I suppose the next iteration would be to create a `/login` route which would take the auth header, and if valid then create a token with a specific lifetime and store it in the DB and set it as a cookie, and validate the cookie on every request.